homeassistant/components/steam_online/config_flow.py

```diff
@@ -121,54 +129,50 @@ async def async_step_init(
     ) -> FlowResult:
         """Manage Steam options."""
         if user_input is not None:
-            await self.hass.config_entries.async_unload(self.entry.entry_id)
-            for _id in self.options[CONF_ACCOUNTS]:
-                if _id not in user_input[CONF_ACCOUNTS] and (
-                    entity_id := er.async_get(self.hass).async_get_entity_id(
-                        Platform.SENSOR, DOMAIN, f"sensor.steam_{_id}"
-                    )
-                ):
-                    er.async_get(self.hass).async_remove(entity_id)
             channel_data = {
                 CONF_ACCOUNTS: {
                     _id: name
                     for _id, name in self.options[CONF_ACCOUNTS].items()
                     if _id in user_input[CONF_ACCOUNTS]
                 }
             }
+            reg = dr.async_get(self.hass)
             await self.hass.config_entries.async_reload(self.entry.entry_id)
+            for _id in self.options[CONF_ACCOUNTS].keys():
```

Comment:
Keys is the default.

```suggestion
            for _id in self.options[CONF_ACCOUNTS]:
```